Do-ItCore/Organization/DoItOrganizationManager.swift

```diff
@@ -6,14 +6,59 @@
 //  Copyright © 2018 The Swifter Picker-Uppers. All rights reserved.
 //
 
+import Foundation
+
 public final class DoItOrganizationManager {
     var organizationSettings: DoItOrganizationSettings
 
     init(organizationSettings: DoItOrganizationSettings) {
         self.organizationSettings = organizationSettings
     }
 
-    func organize(_ doIts: [DoIt]) -> [DoIt] {
-        fatalError("not implemented")
+    func organize(_ doIts: [DoIt]) -> [(String, [DoIt])] {
+        var beingOrganized = doIts
+        let filterAlg = DoItsFilter()
+        for filterType in organizationSettings.filterSetting {
+            beingOrganized = filterAlg.filter(beingOrganized, filterType: filterType)
+        }
+        let sortAlg = DoItSort()
+        var groupedDoIts = groupDoIts(beingOrganized)
+        if let sort = organizationSettings.sortSetting {
+            for index in 0...groupedDoIts.count - 1 {
+                groupedDoIts[index].1 = sortAlg.sortBy(setting: sort,
+                                                       unsortedList: groupedDoIts[index].1)
+            }
+        }
+        return groupedDoIts
+    }
+
+    private func groupDoIts(_ doIts: [DoIt]) -> [(String, [DoIt])] {
+        let groupAlg = DoItGroup()
+        var groupedDoIts = [(String, [DoIt])]()
+        switch organizationSettings.groupingSetting {
+        case .dueDate:
+            let dateFormatter = DateFormatter()
```

Comment:
DateFormatter is a fairy expensive class to create; configure it once and store it as an instance variable.
Swift's `lazy` property modifier would be a good fit.